<!DOCTYPE html>
<html lang="ja">
<head>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({
          google_ad_client: "ca-pub-6300064900388375",
          enable_page_level_ads: true
     });
</script>
<!-- Google Tag Manager -->
<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-NZM8CKQ');</script>
<!-- End Google Tag Manager -->
<style type="text/css">
@font-face { 
	font-family: 'Noto Sans JP';
	src: url('https://nowonbun.github.io/nowonbunfont/NotoSansJP-Regular.otf?version=1') format('opentype');
	font-style:normal;
	font-weight:normal;
}
body{
	font-family: 'Noto Sans JP', 'Arial',sans-serif!important;
}
</style>
<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width,initial-scale=1.0,minimum-scale=1.0,maximum-scale=1.0,user-scalable=no" />
<meta name="theme-color" content="#29343a" />
<link rel="shortcut icon" href="./img/favicon.ico" type="image/x-icon" />
<link rel="icon" href="/favicon.ico" type="image/x-icon" />

<meta name="keywords" content="Study / C#" />
<meta name="description" content="<![CDATA[こんにちは。明月です。今日は例外処理について勉強します。例外処理はキーワード名をみればエラーを解決するキーワードみたいですが実はそうではありません。大きくみると全域の分岐というの意味が近いと思われます。この全域の分岐はエラー処理するところでよく使われるので例外処理と呼ばれるかも知れません。try〜catch、throw例外処理はプロセス処理順「try」ブロックの内で「exception」クラスタイプを「throw」から投げれば「catch」で受け取ります。上の例をみると「if」文と差がなさそうですね。でも「try〜catch,throw」は「try」ブロックの内で「throw」を発生することを「catch」で受け取るものなので「ｔｒｙ」のブロックのメソッド内で「throw」を投げても「catch」で受け取ります。そして継承関係でも「try〜catch、throw」ができます。すなわち親クラスで「exception」を「throw」で投げたら派生クラスの「try〜catch」で受け取ることができます。ここまで確認すると確かに一般分岐と差がありますね。一般分岐の場合は条件によって処理プロセスが変わりますが、「try〜catch、throw」の場合は「try」ブロックの内で「throw」を投げると「catch」にで受け取ることで処理になります。finally「try〜catch」の場合は処理中で投げる形態だから「throw」の次のレベルソースが実行しないです。そうことになると必ず実行すべきなソースレベルがある場合に論理的なエラーがなる可能性があります。上の例を見ると「method」で「exception」が発生すれば「i++」は実行しないので無限ループになります。(実は例みたいに極端的に実装しない。)それでエラーが発生(throw)しても無限ループにならないように「finally」キーワードを使います。「finally」の場合は「try」のブロックの内で返却「return」しても必ず処理するブロックです。なのでプログラムリソースを解除するところでよく使いますね。]]>" />
<meta name="author" content="nowonbun@gmail.com" />
<meta name="distribution" content="nowonbun@gmail.com" />
<meta name="copyright" content="nowonbun@gmail.com" />
<meta name="title" content="明月の開発ストーリ :: [C# スタディ - 23] 例外処理(try〜catch,throwそしてfinally)" />
<link rel="canonical" href="https://www.nowonbun.com/24.html" />
<meta name="robots" content="ALL" />

<meta property="og:type" content="website" />
<meta property="og:title" content="明月の開発ストーリ :: [C# スタディ - 23] 例外処理(try〜catch,throwそしてfinally)" />
<meta property="og:description" content="<![CDATA[こんにちは。明月です。今日は例外処理について勉強します。例外処理はキーワード名をみればエラーを解決するキーワードみたいですが実はそうではありません。大きくみると全域の分岐というの意味が近いと思われます。この全域の分岐はエラー処理するところでよく使われるので例外処理と呼ばれるかも知れません。try〜catch、throw例外処理はプロセス処理順「try」ブロックの内で「exception」クラスタイプを「throw」から投げれば「catch」で受け取ります。上の例をみると「if」文と差がなさそうですね。でも「try〜catch,throw」は「try」ブロックの内で「throw」を発生することを「catch」で受け取るものなので「ｔｒｙ」のブロックのメソッド内で「throw」を投げても「catch」で受け取ります。そして継承関係でも「try〜catch、throw」ができます。すなわち親クラスで「exception」を「throw」で投げたら派生クラスの「try〜catch」で受け取ることができます。ここまで確認すると確かに一般分岐と差がありますね。一般分岐の場合は条件によって処理プロセスが変わりますが、「try〜catch、throw」の場合は「try」ブロックの内で「throw」を投げると「catch」にで受け取ることで処理になります。finally「try〜catch」の場合は処理中で投げる形態だから「throw」の次のレベルソースが実行しないです。そうことになると必ず実行すべきなソースレベルがある場合に論理的なエラーがなる可能性があります。上の例を見ると「method」で「exception」が発生すれば「i++」は実行しないので無限ループになります。(実は例みたいに極端的に実装しない。)それでエラーが発生(throw)しても無限ループにならないように「finally」キーワードを使います。「finally」の場合は「try」のブロックの内で返却「return」しても必ず処理するブロックです。なのでプログラムリソースを解除するところでよく使いますね。]]>" />
<meta property="og:image" content="https://www.nowonbun.com/img/nowonbuntistory1.png" />
<meta property="og:url" content="https://www.nowonbun.com/24.html" />
<meta property="og:site_name" content="明月の開発ストーリ" />

<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:site" content="@nowonbun" />
<meta name="twitter:creator" content="@nowonbun" />
<meta name="twitter:title" content="明月の開発ストーリ :: [C# スタディ - 23] 例外処理(try〜catch,throwそしてfinally)" />
<meta name="twitter:description" content="<![CDATA[こんにちは。明月です。今日は例外処理について勉強します。例外処理はキーワード名をみればエラーを解決するキーワードみたいですが実はそうではありません。大きくみると全域の分岐というの意味が近いと思われます。この全域の分岐はエラー処理するところでよく使われるので例外処理と呼ばれるかも知れません。try〜catch、throw例外処理はプロセス処理順「try」ブロックの内で「exception」クラスタイプを「throw」から投げれば「catch」で受け取ります。上の例をみると「if」文と差がなさそうですね。でも「try〜catch,throw」は「try」ブロックの内で「throw」を発生することを「catch」で受け取るものなので「ｔｒｙ」のブロックのメソッド内で「throw」を投げても「catch」で受け取ります。そして継承関係でも「try〜catch、throw」ができます。すなわち親クラスで「exception」を「throw」で投げたら派生クラスの「try〜catch」で受け取ることができます。ここまで確認すると確かに一般分岐と差がありますね。一般分岐の場合は条件によって処理プロセスが変わりますが、「try〜catch、throw」の場合は「try」ブロックの内で「throw」を投げると「catch」にで受け取ることで処理になります。finally「try〜catch」の場合は処理中で投げる形態だから「throw」の次のレベルソースが実行しないです。そうことになると必ず実行すべきなソースレベルがある場合に論理的なエラーがなる可能性があります。上の例を見ると「method」で「exception」が発生すれば「i++」は実行しないので無限ループになります。(実は例みたいに極端的に実装しない。)それでエラーが発生(throw)しても無限ループにならないように「finally」キーワードを使います。「finally」の場合は「try」のブロックの内で返却「return」しても必ず処理するブロックです。なのでプログラムリソースを解除するところでよく使いますね。]]>" />
<meta property="twitter:image" content="https://www.nowonbun.com/img/nowonbuntistory1.png" />
<meta name="google-site-verification" content="qJOjK0jRyJPKYHFBllVB-mrqZ9_-2Q_3DJxZZvKckFA" />

<title>明月の開発ストーリ :: [C# スタディ - 23] 例外処理(try〜catch,throwそしてfinally)</title>
<link rel="stylesheet" href="//stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous" />
<link rel="stylesheet" type="text/css" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.2/css/all.css" />
<link rel="stylesheet" type="text/css" href="//maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css" />
<link rel="stylesheet" type="text/css" href="//cdnjs.cloudflare.com/ajax/libs/toastr.js/latest/css/toastr.min.css" />
<link rel="stylesheet" type="text/css" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.13.1/styles/github.min.css" />
<link rel="stylesheet" type="text/css" href="./css/highlight.init.css" />
<link rel="stylesheet" type="text/css" href="./css/loader.css" />
<link rel="stylesheet" type="text/css" href="./css/common.css" />
<link rel="stylesheet" type="text/css" href="./css/post.css" />

<script src="//code.jquery.com/jquery-3.3.1.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js" integrity="sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1" crossorigin="anonymous"></script>
<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"></script>

<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/toastr.js/latest/toastr.min.js"></script>
<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/moment.js/2.24.0/moment.min.js"></script>
<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.js"></script>
<script type="text/javascript" src="./js/common.js"></script>

<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>

</head>
<body>
<!-- Google Tag Manager (noscript) -->
<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-NZM8CKQ"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
<!-- End Google Tag Manager (noscript) -->
	<div class="loader off"></div>
	<section class="loader-layout off"></section>
	<nav class="navbar navbar-expand-lg nav-costom">
		<button class="menu-toggle" type="button">MENU</button>
		<a class="navbar-brand navbar-brand-flex-left" href="./" style="color: #000;">明月の開発ストーリ</a>
	</nav>
	<aside class="leftside">
		<h5 class="side-header">
			<img src="https://t1.daumcdn.net/cfile/tistory/1935C94C505D9F8B13">明月の開発ストーリ
		</h5>
		<div class="side-menu bs-glyphicons">
			<ul class="bs-glyphicons-list">
				<li onclick="location.href='./';"><span class="fa fa-home" aria-hidden="true" title="Home"></span> <span>Home</span></li>
				<li><span class="fa fa-minus" aria-hidden="true" title="..."></span><span>...</span></li>
				<li><span class="fa fa-minus" aria-hidden="true" title="..."></span><span>...</span></li>
				<li><span class="fa fa-minus" aria-hidden="true" title="..."></span><span>...</span></li>
				<li class="menu-close"><span class="fa fa-times" aria-hidden="true" title="Close"></span> <span>Close</span></li>
				<li class="menu-close-off"><span class="fa fa-minus" aria-hidden="true" title="..."></span> <span>...</span></li>
			</ul>
		</div>
		<div class="form-inline my-0 search-custom side-search" style="padding: 5px;">
			<input class="form-control mr-2 search-text" type="search" placeholder="Search" aria-label="Search" style="width: 65%">
			<button class="btn btn-outline-success my-0 search-btn" type="submit">Search</button>
		</div>
		<div class="side-list">
			<ul class="side-nav"><li class=""><a class="link_item link-item-collapse category-item" href="javascript:void(0)">Devlopment note<span class="category-item-count">(14)</span><span class="fa fa-chevron-up pull-right"></span></a><ul class="sub_category_list"><li class=""><a class="link_sub_item category-item" data-code="0101" href="dev_clanguage.html">C, C++, MFC<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0102" href="dev_csharp.html">C#<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0103" href="dev_java.html">Java<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0104" href="dev_javascript.html">Javascript, Jquery, Css<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0105" href="dev_window.html">Window<span class="category-item-count">(2)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0106" href="dev_linux.html">Linux<span class="category-item-count">(12)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0107" href="dev_etc.html">Etc.<span class="category-item-count">(0)</span></a></li></ul></li><li class=""><a class="link_item link-item-collapse category-item" href="javascript:void(0)">Study<span class="category-item-count">(91)</span><span class="fa fa-chevron-up pull-right"></span></a><ul class="sub_category_list"><li class=""><a class="link_sub_item category-item" data-code="0201" href="study_clanguage.html">C, C++, MFC<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0202" href="study_csharp.html">C#<span class="category-item-count">(39)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0203" href="study_java.html">Java<span class="category-item-count">(52)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0204" href="study_javascript.html">Javascript, Jquery, Css<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0205" href="study_php.html">PHP<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0206" href="study_python.html">Python<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0207" href="study_database.html">Database<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0208" href="study_design_pattern.html">Design pattern<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_sub_item category-item" data-code="0209" href="study_algorithm.html">Algorithm<span class="category-item-count">(0)</span></a></li></ul></li><li class=""><a class="link_item link-item-collapse category-item" data-code="0300" href="opensource.html">Open source<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_item link-item-collapse category-item" data-code="0400" href="project.html">Project<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_item link-item-collapse category-item" data-code="0500" href="experience.html">Experience<span class="category-item-count">(0)</span></a></li><li class=""><a class="link_item link-item-collapse category-item" data-code="0600" href="other.html">Other<span class="category-item-count">(0)</span></a></li>
			</ul>
		</div>
	</aside>
	<section class="menu-back-layout menu-close off"></section>
	<div class="container-fluid main-container">
		<!-- 메인 -->
		<article class="entry post">
			<div class="titleArea">
				<div class="title" id="article_title">
					<h1 id="titleTxt">[C# スタディ - 23] 例外処理(try〜catch,throwそしてfinally)</h1>
				</div>
				<hr class="titileHr">
				<div class="categoryArea">
					<a href="./?category=0202">Study / C#</a>    作成日付 : 2019/07/15 02:25:26   修正日付 : 2019/07/15 02:26:17
				</div>
			</div>
			<div class="article">
				<!-- 일본 반응형 1 -->
				<ins class="adsbygoogle adsens_responsive" style="display:inline-block" data-ad-client="ca-pub-6300064900388375" data-ad-slot="6281046165"></ins>
				<script>(adsbygoogle = window.adsbygoogle || []).push({});</script>
				<div class="tt_article_useless_p_margin" id="article_contents">
 
 
  <p>こんにちは。明月です。</p> 
  <p><br></p> 
  <p>今日は例外処理について勉強します。例外処理はキーワード名をみればエラーを解決するキーワードみたいですが実はそうではありません。大きくみると全域の分岐というの意味が近いと思われます。</p> 
  <p>この全域の分岐はエラー処理するところでよく使われるので例外処理と呼ばれるかも知れません。</p> 
  <p><br></p> 
  <h2><b><span>try〜catch、throw</span></b></h2> 
  <p>例外処理はプロセス処理順「try」ブロックの内で「Exception」クラスタイプを「throw」から投げれば「catch」で受け取ります。</p> 
  <pre><code class="cs" data-type="">try
{
  throw new Exception();//Exceptionをcatchに投げる。 
}
catch(Exception e)
{
  //Exceptionを受け取る。
}
</code></pre> 
  <pre><code class="cs" data-type="">using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace exceptionEx
{
  class Program
  {
    static void Main(string[] args)
    {
      try
      {
        //tryのブロック
        String data = "ExceptionEx";
        //catchにExceptionを投げる
        throw new Exception("ERROR WORLD");
        Console.WriteLine(data);
      }
      //throwからExceptionを受け取りました。
      catch (Exception e)
      {
        Console.WriteLine(e.Message);
      }

      Console.WriteLine("Press Any Key...");
      Console.ReadLine();
    }
  }
}
</code></pre> 
  <p style="text-align:left;"><img src="./contents/24/83_001.jpg" data-filename="001.jpg" style="width: 600px;"></p> 
  <p>上の例をみると「if」文と差がなさそうですね。でも「try〜catch,throw」は「try」ブロックの内で「throw」を発生することを「catch」で受け取るものなので「ｔｒｙ」のブロックのメソッド内で「throw」を投げても「catch」で受け取ります。</p> 
  <pre><code class="cs" data-type="">﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace exceptionEx
{
  class Program
  {
    static void Main(string[] args)
    {
      try
      {
        //tryのブロック
        String data = "ExceptionEx";
        Process();
        Console.WriteLine(data);
      }
      //throwからExceptionを受け取りました。
      catch (Exception e)
      {
        Console.WriteLine(e.Message);
      }

      Console.WriteLine("Press Any Key...");
      Console.ReadLine();
    }
    //メソッド内で「throw」を投げる。
    static void Process()
    {
      //catchにExceptionを投げる
      throw new Exception("ERROR Method WORLD");
    }
  }
}
</code></pre> 
  <p style="text-align:left;"><img src="./contents/24/84_002.jpg" data-filename="002.jpg" style="width: 600px;"></p> 
  <p>そして継承関係でも「try〜catch、throw」ができます。すなわち親クラスで「Exception」を「throw」で投げたら派生クラスの「try〜catch」で受け取ることができます。</p> 
  <pre><code class="cs" data-type="">﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace exceptionEx
{
  class ParentClass
  {
    //親クラスから派生クラスに「throw」を投げるメソッド
    public virtual void Process()
    {
      //catchにExceptionを投げる
      throw new Exception("ERROR Abstract WORLD");
    }
  }
  class ChildClass : ParentClass
  {
    public override void Process()
    {
      try
      {
        //tryのブロック
        String data = "ExceptionEx";
        base.Process();
        Console.WriteLine(data);
      }
      //throwからExceptionを受け取りました。
      catch (Exception e)
      {
        Console.WriteLine(e.Message);
      }
    }
  }
  class Program
  {
    static void Main(string[] args)
    {
      //「ChildClass」を割り当て
      ChildClass child = new ChildClass();
      child.Process();

      Console.WriteLine("Press Any Key...");
      Console.ReadLine();
    }
  }
}
</code></pre> 
  <p style="text-align:left;"><img src="./contents/24/85_003.jpg" data-filename="003.jpg" style="width: 600px;"></p> 
  <p>ここまで確認すると確かに一般分岐と差がありますね。一般分岐の場合は条件によって処理プロセスが変わりますが、「try〜catch、throw」の場合は「try」ブロックの内で「throw」を投げると「catch」にで受け取ることで処理になります。</p> 
  <p><br></p> 
  <h2><b><span>finally</span></b></h2> 
  <p>「try〜catch」の場合は処理中で投げる形態だから「throw」の次のレベルソースが実行しないです。そうことになると必ず実行すべきなソースレベルがある場合に論理的なエラーがなる可能性があります。</p> 
  <pre><code class="cs" data-type="">using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace exceptionEx
{
  class ExceptionExClass
  {
    public void Process()
    {
      int i = 0;
      while (i &lt; 10)
      {
        //無限ループになる。
        try
        {
          //「method」メソッドで「Exception」が発生
          method();
          //それで「i」の値が増減しない
          i++;
        }
        catch (Exception e)
        {
          Console.WriteLine(e.Message);
        }
      }
    }
    public void method()
    {
      throw new Exception("ERROR");
    }
  }
  class Program
  {
    static void Main(string[] args)
    {
      //「ExceptionExClass」を割り当て
      ExceptionExClass child = new ExceptionExClass();
      child.Process();

      Console.WriteLine("Press Any Key...");
      Console.ReadLine();
    }
  }
}
</code></pre> 
  <p>上の例を見ると「method」で「Exception」が発生すれば「i++」は実行しないので無限ループになります。(実は例みたいに極端的に実装しない。)</p> 
  <p>それでエラーが発生(throw)しても無限ループにならないように「finally」キーワードを使います。</p> 
  <pre><code class="cs" data-type="">﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace exceptionEx
{
  class ExceptionExClass
  {
    public void Process()
    {
      int i = 0;
      while (i &lt; 10)
      {
        //無限ループになる。
        try
        {
          //「method」メソッドで「Exception」が発生
          method();
        }
        catch (Exception e)
        {
          Console.WriteLine(e.Message);
        }
        finally
        {
          //「Exception」が発生しても必ず実行する「finally」
          i++;
        }
      }
    }
    public void method()
    {
      throw new Exception("ERROR");
    }
  }
  class Program
  {
    static void Main(string[] args)
    {
      //「ExceptionExClass」を割り当て
      ExceptionExClass child = new ExceptionExClass();
      child.Process();

      Console.WriteLine("Press Any Key...");
      Console.ReadLine();
    }
  }
}
</code></pre> 
  <p style="text-align:left;"><img src="./contents/24/81_004.jpg" data-filename="004.jpg" style="width: 600px;"></p> 
  <p>「finally」の場合は「try」のブロックの内で返却「return」しても必ず処理するブロックです。なのでプログラムリソースを解除するところでよく使いますね。</p> 
  <pre><code class="cs" data-type="">﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace exceptionEx
{
  class ExceptionExClass
  {
    public void Process()
    {
      //無限ループになる。
      try
      {
        //返却する。
        return;
      }
      catch (Exception e)
      {
        Console.WriteLine(e.Message);
      }
      finally
      {
        //返却しても必ず実行する「finally」
        Console.WriteLine("finally!!!!!!");
      }
    }
  }
  class Program
  {
    static void Main(string[] args)
    {
      //「ExceptionExClass」を割り当て
      ExceptionExClass child = new ExceptionExClass();
      child.Process();

      Console.WriteLine("Press Any Key...");
      Console.ReadLine();
    }
  }
}
</code></pre> 
  <p style="text-align:left;"><img src="./contents/24/82_005.jpg" data-filename="005.jpg" style="width: 600px;"></p>
 
</div>
				<!-- 일본 반응형 2 -->
				<ins class="adsbygoogle adsens_responsive" style="display:inline-block" data-ad-client="ca-pub-6300064900388375" data-ad-slot="5897902785"></ins>
				<script>(adsbygoogle = window.adsbygoogle || []).push({});</script>
				<div class="list-meta ie-dotum">
					<span class="timeago ff-h dt-published tag-column" id="article_tag"> <a href=./search.html?query=C%23>#C#</a> </span>
				</div>
				<div class="other-category">
					<div class="list-title">「<a href="./?category=0202">Study / C#</a>」の他投稿</div>
					<ul><li><a href='https://www.nowonbun.com/40.html'>[C# スタディ - 39] Thread(スレッド)</a><span>2019/07/24 00:57:35</span></li><li><a href='https://www.nowonbun.com/39.html'>[C# スタディ - 38] 「IDisposable」と「using」</a><span>2019/07/23 00:05:40</span></li><li><a href='https://www.nowonbun.com/38.html'>[C# スタディ - 37] ファイルシステム(File System) - FileStream、 Encoding、 Serializable</a><span>2019/07/22 23:45:05</span></li><li><a href='https://www.nowonbun.com/37.html'>[C# スタディ - 36] ファイルシステム(File System) - FileInfo,DirectoryInfo</a><span>2019/07/22 23:30:17</span></li><li><a href='https://www.nowonbun.com/36.html'>[C# スタディ - 35] MSDN(Microsoft Developer Network)</a><span>2019/07/22 23:15:42</span></li><li><a href='https://www.nowonbun.com/35.html'>[C# スタディ - 34] Attribute(アトリビュート)</a><span>2019/07/20 02:27:23</span></li><li><a href='https://www.nowonbun.com/34.html'>[C# スタディ - 33] Reflection(リフレクション)</a><span>2019/07/20 02:22:03</span></li><li><a href='https://www.nowonbun.com/33.html'>[C# スタディ - 32] Objectタイプ、varタイプ(匿名タイプ)、dynamicタイプ</a><span>2019/07/18 22:50:16</span></li><li><a href='https://www.nowonbun.com/32.html'>[C# スタディ - 31] Partial Type(クラス分割)、拡張メソッド</a><span>2019/07/18 20:22:16</span></li><li><a href='https://www.nowonbun.com/31.html'>[C# スタディ - 30] Lamda(ラムダ)</a><span>2019/07/17 23:06:42</span></li><li><a href='https://www.nowonbun.com/30.html'>[C# スタディ - 29] LINQ(リンク)-3 (Enumerableクラス)</a><span>2019/07/17 20:57:00</span></li><li><a href='https://www.nowonbun.com/29.html'>[C# スタディ - 28] LINQ(リンク)-2 (メソッド式)</a><span>2019/07/16 22:40:03</span></li><li><a href='https://www.nowonbun.com/28.html'>[C# スタディ - 27] LINQ(リンク)-1</a><span>2019/07/16 20:41:27</span></li><li><a href='https://www.nowonbun.com/27.html'>[C# スタディ - 26] event(イベント)</a><span>2019/07/16 00:59:34</span></li><li><a href='https://www.nowonbun.com/26.html'>[C# スタディ - 25] delegate(デリゲート)</a><span>2019/07/16 00:48:03</span></li></ul>
				</div>
				<div class="recently-post">
					<div class="list-title">最新投稿</div>
					<ul><li><a href='https://www.nowonbun.com/105.html'>[CentOS] FTPインストールする方法(vsftpd)</a><span>2019/11/01 07:29:57</span></li><li><a href='https://www.nowonbun.com/104.html'>[CentOS] SAMBAインストール方法</a><span>2019/10/30 07:38:57</span></li><li><a href='https://www.nowonbun.com/103.html'>[CentOS] PHPインストールする方法</a><span>2019/10/29 20:09:03</span></li><li><a href='https://www.nowonbun.com/102.html'>[CentOS] ApacheとTomcatの連携</a><span>2019/10/29 07:28:33</span></li><li><a href='https://www.nowonbun.com/101.html'>[CentOS] Tomcat インストール方法</a><span>2019/10/28 22:38:56</span></li><li><a href='https://www.nowonbun.com/100.html'>[CentOS] Java インストール</a><span>2019/10/28 00:22:32</span></li><li><a href='https://www.nowonbun.com/99.html'>[CentOS] apacheをインストールする方法</a><span>2019/10/25 21:43:27</span></li><li><a href='https://www.nowonbun.com/98.html'>[Linux] ディスク使用率チェック(df, du, lsof)とdfバグを解決方法</a><span>2019/10/24 21:46:31</span></li><li><a href='https://www.nowonbun.com/97.html'>[CentOS] Wheel設定</a><span>2019/10/23 23:53:40</span></li><li><a href='https://www.nowonbun.com/96.html'>[CentOS] システムの文字セット変更</a><span>2019/10/21 16:58:39</span></li><li><a href='https://www.nowonbun.com/95.html'>[CentOS] Vim編集ツールをインストールする方法</a><span>2019/10/21 00:24:44</span></li><li><a href='https://www.nowonbun.com/94.html'>[Linux] Putty使用方法</a><span>2019/10/20 00:42:54</span></li><li><a href='https://www.nowonbun.com/93.html'>[Window] Apacheをインストール方法、サービスに登録する方法。</a><span>2019/10/18 07:36:51</span></li><li><a href='https://www.nowonbun.com/92.html'>[Javaスタディ- 52] Java servletでインスタンスを初期する方法</a><span>2019/10/17 07:15:48</span></li><li><a href='https://www.nowonbun.com/91.html'>[Javaスタディ- 51] Spring web frameworkで発生する文字化けのEncoding設定</a><span>2019/10/16 07:32:55</span></li></ul>
				</div>
			</div>
		</article>
		<script type="text/javascript" src="./js/postscreen.js"></script>
	</div>
	<footer>
		<div class="col-lg-12 col-md-12 col-sm-12 col-xs-12 footer-bottom">
			<p>Copyright © 2019 www.nowonbun.com(Email : <a href="mailto:nowonbun@gmail.com" target="_blank">nowonbun@gmail.com</a>)</p>
		</div>
	</footer>
	<div id="ImgModal" class="modal">
		<span class="close img-type">×</span> <img class="modal-content">
	</div>
	<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.13.1/highlight.min.js"></script>
	<script src="./js/highlightjs-line-numbers.js"></script>
	<script>hljs.initHighlightingOnLoad();</script>
	<script>hljs.initLineNumbersOnLoad();</script>
</body>
</html>
